src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/ProPlanControls/ProPlanControls.jsx

```diff
@@ -0,0 +1,107 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useForm } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+
+import {
+  useAccountDetails,
+  useAvailablePlans,
+  usePlanData,
+} from 'services/account'
+import { useProPlans } from 'shared/utils/billing'
+import {
+  getDefaultValuesProUpgrade,
+  getSchema,
+  MIN_NB_SEATS_PRO,
+} from 'shared/utils/upgradeForm'
+import TextInput from 'ui/TextInput'
+
+import BillingControls from './BillingControls'
+import TotalBanner from './TotalBanner'
+import UserCount from './UserCount'
+
+import { useUpgradeControls } from '../hooks'
+import UpdateButton from '../UpdateButton'
+
+function ProPlanControls() {
+  const { provider, owner } = useParams()
+  const { data: accountDetails } = useAccountDetails({ provider, owner })
+  const { data: plans } = useAvailablePlans({ provider, owner })
+  const { data: planData } = usePlanData({ owner, provider })
+  const { proPlanYear } = useProPlans({ plans })
+  const { upgradePlan } = useUpgradeControls()
+
+  const trialStatus = planData?.plan?.trialStatus
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isValid, errors },
+    setValue,
+    getValues,
+  } = useForm({
```

Comment:
This extracts form data. I could put defaultValues into it's own variables too, no biggie. But yeah, using this is easier to read which are responsibilities of react-form and not the controls hook